voila/voilite/app.py

```diff
@@ -0,0 +1,180 @@
+    }
+
+    @default('log_level')
+    def _default_log_level(self):
+        return logging.INFO
+
+    @default('root_dir')
+    def _default_root_dir(self):
+        if self.notebook_path:
+            return os.path.dirname(os.path.abspath(self.notebook_path))
+        else:
+            return os.getcwd()
+
+    def initialize(self, argv=None):
+        super().initialize(argv)
+        if len(self.extra_args) == 1:
+            arg = self.extra_args[0]
+
+            if not self.notebook_path:
+                if os.path.isdir(arg):
+                    self.root_dir = arg
+                elif os.path.isfile(arg):
+                    self.notebook_path = arg
+                else:
+                    raise ValueError(
+                        'argument is neither a file nor a directory: %r' % arg
+                    )
+        elif len(self.extra_args) != 0:
+            raise ValueError(
+                'provided more than 1 argument: %r' % self.extra_args
+            )
+        self.voilite_configuration = VoilaConfiguration(parent=self)
+
+    def copy_static_files(self, page_config: Dict) -> None:
+
+        lite_static_path = os.path.join(
+            os.path.dirname(os.path.realpath(__file__)), 'static'
+        )
+        dest_static_path = os.path.join(os.getcwd(), 'voila')
+        if os.path.isdir(dest_static_path):
+            shutil.rmtree(dest_static_path)
+        copy_tree(
+            os.path.join(lite_static_path, 'voila'),
+            os.path.join(dest_static_path, 'static'),
+        )
+        shutil.move(
+            os.path.join(dest_static_path, 'static', 'pypi'),
+            os.path.join(dest_static_path, 'build', 'pypi'),
+        )
+        shutil.copyfile(
+            os.path.join(lite_static_path, 'services.js'),
+            os.path.join(dest_static_path, 'services.js'),
+        )
+
+        # Copy extension files
+        labextensions_path = jupyter_path('labextensions')
+        federated_extensions = page_config.get('federated_extensions', [])
+        roots = tuple(
+            os.path.abspath(os.path.expanduser(p)) + os.sep
+            for p in labextensions_path
+        )
+        dest_extension_path = os.path.join(
+            os.getcwd(), 'voila', 'labextensions'
+        )
+        if os.path.isdir(dest_extension_path):
+            shutil.rmtree(dest_extension_path)
+        for extension in federated_extensions:
+            for root in roots:
+                name = extension['name']
+                full_path = os.path.join(root, name)
+                if os.path.isdir(full_path):
+                    copy_tree(
+                        full_path, os.path.join(dest_extension_path, name)
+                    )
+
+        # Copy themes files
+        all_themes = find_all_lab_theme()
+        for theme in all_themes:
+            theme_dst = os.path.join(
+                dest_static_path, 'static', 'themes', theme[0]
+            )
+            copy_tree(theme[1], theme_dst)
```

Comment:
We might be able to get rid of all this if we make it a JupyterLite addon?

Reply:
I'm not very familiar with JupyterLite, does that mean `Voila` will depend on `JupyterLite`?

Reply:
The xeus python lite kernel is a good example of a JupyterLite addon: https://github.com/jupyterlite/xeus-python-kernel/blob/main/jupyterlite_xeus_python/env_build_addon.py

Reply:
We could probably make `voilite` and `voila` two specific Python packages so that we don't make voila depend on jupyterlite

Reply:
Frankly, I'm not a fan of splitting `voila/voilite`, even at the command level.
I would love to be able to switch seamlessly between the two versions in the same `voila` command. It would be even better if we can mix the two types together on the same page. For example, when we serve a directory of notebooks we can config (by `voila.json`, `voila.py`,...) which ones will be served by the normal `voila`, and which one by the lite version.